hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/TenantControl.java

```diff
@@ -43,18 +41,19 @@
     /**
      * Establish this tenant's thread-local context
      * Particular TenantControl implementation will control the details of how
-     * createRequestScope parameter is handled, but in general,
-     * if createRequestScope = false, only ClassLoader is set up,
-     * if createRequestScope = true, in addition to ClassLoader,
-     * other things like invocation, EJB/JPA/CDI context should be set up as well
      *
-     * In other words, if only app class needs to be resolved, set createRequestScope to false
-     * If actually calling into user's code, set createRequestScope to true
-     *
-     * @param createRequestScope whether to create CDI request scope for this context
      * @return handle to be able to close the tenant's scope.
      */
-    Closeable setTenant(boolean createRequestScope);
+    Closeable setTenant();
+
+    /**
+     * To be called when Hazelcast object is created
+     * @param destroyEventContext hook to decouple any Hazelcast object when the tenant is destroyed,
+     * This is used, for example, to delete all associated caches from the application when
+     * it gets undeployed, so there are no ClassCastExceptions afterwards.
+     * Cannot be null. This is a functional interface, so no-op lambda can be used instead.
+     */
+    void distributedObjectCreated(DestroyEventContext destroyEventContext);
```

Comment:
You can also add `Nonnull` annotation here then, also add `Nonnull` or `Nullable` for other methods, parameters and return values.

Reply:
done